### ISSUE TITLE
Move `formatEnabledInEnv` to a better place

### ISSUE THREAD

**Issue by jvican** [sbt/sbt#3371]
## problem

**There's not a good, intuitive way of figuring out if the underlying logger/shell supports colours**.

<s>sbt 1.0.0-RC2 changed `formatEnabled` to `formatEnabledInEnv`. I don't know what triggered this change, but I don't think it was a good idea... In any case, now that it's already done, we should add ways to maintain source compatibility for all plugins.
  
My idea would be to add a this one-liner:
```scala
@deprecated("Use formatEnabledInEnv instead", "1.0.0")
val formatEnabled: Boolean = formatEnabledInEnv
```
</s>
## expectation
There's an API called `Logger` or `LoggerUtils` or something alike that allows me to easily check this. Or even a global key would suffice.

<s>`ConsoleAppender.formatEnabled` does compile.</s>

## notes
  
IMO, `ConsoleAppender` is not the right place to have all this information. So we should consider locating this **crucial** info in a more direct and visible place, which makes more sense than `ConsoleAppender`, which is not intuitive for those totally unaware of how the underlying log mechanism works (which is most of the people).

sbt version:1.0.0-RC2

**Comment by dwijnand** [comment]
In my opinion, we don't need to support source-compatibility between pre-release versions of sbt.

For source compatibility with sbt 0.13.x, which I think is a very worthy cause, we do have that deprecation in place (the "since" field is wrong and there's a typo in the message though.. cc @Duhemm):

```scala
  @deprecated("Use ConsoleAppenders.formatEnabledInEnv", "0.13.x")
  val formatEnabled = ConsoleAppender.formatEnabledInEnv
```

(https://github.com/sbt/util/blob/v1.0.0-M28/internal/util-logging/src/main/scala/sbt/internal/util/ConsoleAppender.scala#L25-L26)

**Comment by jvican** [comment]
Let me change the main goal of this ticket towards providing a sane API for plugin authors to know if they can use colors or not in their output. I agree than source-compatibility between pre-release versions of sbt is not that of a big deal.

**Comment by dwijnand** [comment]
I agree we should consider making this public API. Currently it's in `sbt.internal.util.ConsoleAppender.formatEnabledInEnv`.

**Comment by eed3si9n** [comment]
What used to be called "Logger" in 0.13 now splits into two separate categories of classes that many are already familiar from log4j etc: 
1. `Logger` that is responsible for generating log events, and 
2. `Appender` that consumes those events.

From the point of view of the plugin (or tasks) it's not knowable whether _all_ of the terminal supports formatting or not because the idea behind sbt server is that the events could be transmitted to another program via JSON.
